js/search_bar.ts

```diff
@@ -0,0 +1,400 @@
+import type { RenderProps } from "@anywidget/types";
+import { html, css, PropertyValueMap } from "lit";
+import { property, query, queryAll } from "lit/decorators.js";
+import { classMap } from "lit/directives/class-map.js";
+
+import { legacyStyles } from "./ipywidgets_styles";
+import { LitWidget } from "./lit_widget";
+import { materialStyles } from "./styles";
+import { loadFonts } from "./utils";
+
+import { TabMode } from "./tab_panel";
+import { unsafeHTML } from "lit/directives/unsafe-html.js";
+
+export interface SearchTab {
+    search: string,
+    results: string[],
+    selected: string,
+    additional_html: string,
+}
+
+
+export interface SearchBarModel {
+    expanded: boolean;
+    tab_index: number;
+    name_address_model: string,
+    lat_lon_model: string,
+    dataset_model: string,
+}
+
+const DEBOUNCE_TIMEOUT = 500;
```

Comment:
Consider moving `DEBOUNCE_TIMEOUT` and `debounce` to utils.ts.

Reply:
Done.